include/emds/emds.h

```diff
@@ -112,9 +113,20 @@ int emds_entry_add(struct emds_dynamic_entry *entry);
  *
  * Triggers the process of storing all data registered to be stored. All data
  * registered either through @ref emds_entry_add function or the
- * @ref EMDS_STATIC_ENTRY_DEFINE macro is stored. Once the data storage is
- * completed, the data should not be changed, and the device should be halted.
+ * @ref EMDS_STATIC_ENTRY_DEFINE macro is stored. It locks all interrupts until
+ * the write is finished. Once the data storage is completed, the data should
+ * not be changed, and the device should be halted. The device must not be
+ * allowed to reboot when operating on a backup supply. Since reboot will
+ * trigger data load from EMDS storage and clear the storage area. The reboot
+ * should be allowed only when the main power supply is available.
```

Comment:
```suggestion
 * not be changed, and the device should be halted. The device must not be
 * allowed to reboot when operating on a backup supply, since reboot will
 * trigger data load from the EMDS storage and clear the storage area. The reboot
 * should only be allowed when the main power supply is available.
```